### PR TITLE
embed dnsmasq_exporter

### PR DESCRIPTION
**Pull request by rfratto**
If you use `examples/docker-compose/docker-compose.integrations.yaml` and `cmd/agent/agent-integrations-config.yaml`, you can test this and verify that it's working.

### CODE REVIEW

File: cmd/agent/agent-integrations-config.yaml [hunk 1]
```diff
@@ -23,7 +23,11 @@ integrations:
     data_source_name: root@(localhost:3306)/
   redis_exporter:
     enabled: true
-    redis_addr: redis:6379
+    redis_addr: localhost:6379
```
Comment (rfratto) [inline cmd/agent/agent-integrations-config.yaml:26]: I changed the address here to work with the docker compose file.

File: docs/configuration-reference.md [hunk 2]
```diff
@@ -2680,3 +2683,71 @@ Full reference of options:
...
+  relabel_configs:
+  - source_labels: [__address__]
+    target_label: instance
+    replacement: dnsmasq-a
+```
+
+Full reference of options:
+
+```yaml
+  # Enables the redis_exporter integration, allowing the Agent to automatically
```
Comment (hoenn) [inline docs/configuration-reference.md:2712]:
```suggestion
  # Enables the dnsmasq_exporter integration, allowing the Agent to automatically
```

File: docs/configuration-reference.md [hunk 3]
```diff
@@ -2680,3 +2683,71 @@ Full reference of options:
...
+  # collect system metrics from the configured redis address
+  [enabled: <boolean> | default = false]
+
+  # Automatically collect metrics from this integration. If disabled,
+  # the redis_exporter integration will be run but not scraped and thus not
```
Comment (hoenn) [inline docs/configuration-reference.md:2717]:
```suggestion
  # the dnsmasq_exporter integration will be run but not scraped and thus not
```

File: docs/configuration-reference.md [hunk 4]
```diff
@@ -2680,3 +2683,71 @@ Full reference of options:
...
+  # remote-written. Metrics for the integration will be exposed at
+  # /integrations/redis_exporter/metrics and can be scraped by an external
```
Comment (hoenn) [inline docs/configuration-reference.md:2719]:
```suggestion
  # /integrations/dnsmasq_exporter/metrics and can be scraped by an external
```

File: pkg/integrations/dnsmasq_exporter/config.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+// DefaultConfig is the default config for dnsmasq_exporter.
+var DefaultConfig Config = Config{
+	DnsmasqAddress: "localhost:53",
+	LeasesPath:     "/var/lib/misc/dnsmasq.leases",
+}
+
+// Config controls the mysqld_exporter integration.
```
Comment (hoenn) [inline pkg/integrations/dnsmasq_exporter/config.go:13]:
```suggestion
// Config controls the dnsmasq_exporter integration.
```

File: docs/configuration-reference.md [hunk 6]
```diff
@@ -2680,3 +2683,71 @@ Full reference of options:
...
+  - source_labels: [__address__]
+    target_label: instance
+    replacement: dnsmasq-a
+```
+
+Full reference of options:
+
+```yaml
+  # Enables the dnsmasq_exporter integration, allowing the Agent to automatically
+  # collect system metrics from the configured redis address
```
Comment (dafydd-t) [inline docs/configuration-reference.md:2713]:
```suggestion
  # collect system metrics from the configured dnsmasq address
```

**Review by dafydd-t — APPROVED** [review]
👌

**Comment by rfratto** [comment]
I promise next time I won't forget to replace a word with the new exporter. Thank you both!

**Pull request merged**